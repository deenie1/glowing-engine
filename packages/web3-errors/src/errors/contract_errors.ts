/*
This file is part of web3.js.

web3.js is free software: you can redistribute it and/or modify
it under the terms of the GNU Lesser General Public License as published by
the Free Software Foundation, either version 3 of the License, or
(at your option) any later version.

web3.js is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU Lesser General Public License for more details.

You should have received a copy of the GNU Lesser General Public License
along with web3.js.  If not, see <http://www.gnu.org/licenses/>.
*/

/* eslint-disable max-classes-per-file */

import { JsonRpcError, TransactionReceipt, HexString } from 'web3-types';
import {
	ERR_CONTRACT,
	ERR_CONTRACT_ABI_MISSING,
	ERR_CONTRACT_EXECUTION_REVERTED,
	ERR_CONTRACT_EVENT_NOT_EXISTS,
	ERR_CONTRACT_INSTANTIATION,
	ERR_CONTRACT_MISSING_ADDRESS,
	ERR_CONTRACT_MISSING_DEPLOY_DATA,
	ERR_CONTRACT_MISSING_FROM_ADDRESS,
	ERR_CONTRACT_REQUIRED_CALLBACK,
	ERR_CONTRACT_RESERVED_EVENT,
	ERR_CONTRACT_RESOLVER_MISSING,
	ERR_CONTRACT_TX_DATA_AND_INPUT,
} from '../error_codes';
import { BaseWeb3Error, InvalidValueError } from '../web3_error_base';

export class Web3ContractError extends BaseWeb3Error {
	public code = ERR_CONTRACT;
	public receipt?: TransactionReceipt;

	public constructor(message: string, receipt?: TransactionReceipt) {
		super(message);

		this.receipt = receipt;
	}
}
export class ResolverMethodMissingError extends BaseWeb3Error {
	public code = ERR_CONTRACT_RESOLVER_MISSING;

	public constructor(public address: string, public name: string) {
		super(`The resolver at ${address} does not implement requested method: "${name}".`);
	}

	public toJSON() {
		return { ...super.toJSON(), address: this.address, name: this.name };
	}
}

export class ContractMissingABIError extends BaseWeb3Error {
	public code = ERR_CONTRACT_ABI_MISSING;

	public constructor() {
		super(
			'You must provide the json interface of the contract when instantiating a contract object.',
		);
	}
}

export class ContractOnceRequiresCallbackError extends BaseWeb3Error {
	public code = ERR_CONTRACT_REQUIRED_CALLBACK;

	public constructor() {
		super('Once requires a callback as the second parameter.');
	}
}

export class ContractEventDoesNotExistError extends BaseWeb3Error {
	public code = ERR_CONTRACT_EVENT_NOT_EXISTS;

	public constructor(public eventName: string) {
		super(`Event "${eventName}" doesn't exist in this contract.`);
	}

	public toJSON() {
		return { ...super.toJSON(), eventName: this.eventName };
	}
}

export class ContractReservedEventError extends BaseWeb3Error {
	public code = ERR_CONTRACT_RESERVED_EVENT;

	public constructor(public type: string) {
		super(`Event "${type}" doesn't exist in this contract.`);
	}

	public toJSON() {
		return { ...super.toJSON(), type: this.type };
	}
}

export class ContractMissingDeployDataError extends BaseWeb3Error {
	public code = ERR_CONTRACT_MISSING_DEPLOY_DATA;

	public constructor() {
		super(`No "data" specified in neither the given options, nor the default options.`);
	}
}

export class ContractNoAddressDefinedError extends BaseWeb3Error {
	public code = ERR_CONTRACT_MISSING_ADDRESS;

	public constructor() {
		super("This contract object doesn't have address set yet, please set an address first.");
	}
}

export class ContractNoFromAddressDefinedError extends BaseWeb3Error {
	public code = ERR_CONTRACT_MISSING_FROM_ADDRESS;

	public constructor() {
		super('No "from" address specified in neither the given options, nor the default options.');
	}
}

export class ContractInstantiationError extends BaseWeb3Error {
	public code = ERR_CONTRACT_INSTANTIATION;
}

/**
 * This class is expected to be set as an `innerError` inside ContractExecutionError
 * The properties would be typically decoded from the `data` if it was encoded according to EIP-838
 */
export class Eip838ExecutionError extends Web3ContractError {
	public readonly name: string;
	public code: number;
	public data?: HexString;
	public errorName?: string;
	public errorSignature?: string;
	public errorArgs?: { [K in string]: unknown };

	public constructor(code: number, message: string, data?: HexString) {
		super(message);
		this.name = this.constructor.name;
		this.code = code;
		this.data = data;
	}

	public setDecodedProperties(
		errorName: string,
		errorSignature?: string,
		errorArgs?: { [K in string]: unknown },
	) {
		this.errorName = errorName;
		this.errorSignature = errorSignature;
		this.errorArgs = errorArgs;
	}
}

/**
 * Used when an error is raised while executing a function inside a smart contract.
 * The data is expected to be encoded according to EIP-848.
 */
export class ContractExecutionError extends Web3ContractError {
	public innerError: Eip838ExecutionError;

	public constructor(rpcError: JsonRpcError) {
		super('Error happened while trying to execute a function inside a smart contract');
		this.code = ERR_CONTRACT_EXECUTION_REVERTED;
		this.innerError = new Eip838ExecutionError(
			rpcError.code,
			rpcError.message,
			rpcError.data as string,
		);
	}
}

export class ContractTransactionDataAndInputError extends InvalidValueError {
	public code = ERR_CONTRACT_TX_DATA_AND_INPUT;

	public constructor(value: { data: HexString | undefined; input: HexString | undefined }) {
		super(
			`data: ${value.data ?? 'undefined'}, input: ${value.input ?? 'undefined'}`,
			'You can\'t have "data" and "input" as properties of a contract at the same time, please use either "data" or "input" instead.',
		);
	}
}
