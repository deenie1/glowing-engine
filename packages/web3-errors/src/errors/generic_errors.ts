/*
This file is part of web3.js.

web3.js is free software: you can redistribute it and/or modify
it under the terms of the GNU Lesser General Public License as published by
the Free Software Foundation, either version 3 of the License, or
(at your option) any later version.

web3.js is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU Lesser General Public License for more details.

You should have received a copy of the GNU Lesser General Public License
along with web3.js.  If not, see <http://www.gnu.org/licenses/>.
*/

/* eslint-disable max-classes-per-file */

import {
	ERR_ABI_ENCODING,
	ERR_FORMATTERS,
	ERR_METHOD_NOT_IMPLEMENTED,
	ERR_OPERATION_ABORT,
	ERR_OPERATION_TIMEOUT,
	ERR_PARAM,
} from '../error_codes';
import { Web3Error } from '../web3_error_base';

export class InvalidNumberOfParamsError extends Web3Error {
	public code = ERR_PARAM;

	public constructor(public got: number, public expected: number, public method: string) {
		super(`Invalid number of parameters for "${method}". Got "${got}" expected "${expected}"!`);
	}

	public toJSON() {
		return {
			...super.toJSON(),
			got: this.got,
			expected: this.expected,
			method: this.method,
		};
	}
}

export class FormatterError extends Web3Error {
	public code = ERR_FORMATTERS;
}

export class MethodNotImplementedError extends Web3Error {
	public code = ERR_METHOD_NOT_IMPLEMENTED;

	public constructor() {
		super("The method you're trying to call is not implemented.");
	}
}

export class OperationTimeoutError extends Web3Error {
	public code = ERR_OPERATION_TIMEOUT;
}

export class OperationAbortError extends Web3Error {
	public code = ERR_OPERATION_ABORT;
}

export class AbiError extends Web3Error {
	public code = ERR_ABI_ENCODING;
}
