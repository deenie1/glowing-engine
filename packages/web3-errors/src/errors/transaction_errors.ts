/*
This file is part of web3.js.

web3.js is free software: you can redistribute it and/or modify
it under the terms of the GNU Lesser General Public License as published by
the Free Software Foundation, either version 3 of the License, or
(at your option) any later version.

web3.js is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU Lesser General Public License for more details.

You should have received a copy of the GNU Lesser General Public License
along with web3.js.  If not, see <http://www.gnu.org/licenses/>.
*/

/* eslint-disable max-classes-per-file */

import { TransactionReceipt } from 'web3-types';
import {
	ERR_RAW_TX_UNDEFINED,
	ERR_TX,
	ERR_TX_CONTRACT_NOT_STORED,
	ERR_TX_NO_CONTRACT_ADDRESS,
	ERR_TX_OUT_OF_GAS,
	ERR_TX_REVERT_INSTRUCTION,
	ERR_TX_REVERT_TRANSACTION,
	ERR_TX_REVERT_WITHOUT_REASON,
	ERR_TX_NOT_FOUND,
} from '../error_codes';
import { Web3Error } from '../web3_error_base';

export class TransactionError<ReceiptType = TransactionReceipt> extends Web3Error {
	public code = ERR_TX;

	public constructor(message: string, public receipt?: ReceiptType) {
		super(message);
	}

	public toJSON() {
		return { ...super.toJSON(), receipt: this.receipt };
	}
}

export class RevertInstructionError extends Web3Error {
	public code = ERR_TX_REVERT_INSTRUCTION;

	public constructor(public reason: string, public signature: string) {
		super(`Your request got reverted with the following reason string: ${reason}`);
	}

	public toJSON() {
		return { ...super.toJSON(), reason: this.reason, signature: this.signature };
	}
}

export class TransactionRevertError extends Web3Error {
	public code = ERR_TX_REVERT_TRANSACTION;

	public constructor(
		public reason: string,
		public signature: string,
		public receipt: TransactionReceipt,
	) {
		super(
			`Transaction has been reverted by the EVM:\n ${JSON.stringify(receipt, undefined, 2)}`,
		);
	}

	public toJSON() {
		return {
			...super.toJSON(),
			reason: this.reason,
			signature: this.signature,
			receipt: this.receipt,
		};
	}
}

export class NoContractAddressFoundError extends TransactionError {
	public constructor(receipt: TransactionReceipt) {
		super("The transaction receipt didn't contain a contract address.", receipt);
		this.code = ERR_TX_NO_CONTRACT_ADDRESS;
	}

	public toJSON() {
		return { ...super.toJSON(), receipt: this.receipt };
	}
}

export class ContractCodeNotStoredError extends TransactionError {
	public constructor(receipt: TransactionReceipt) {
		super("The contract code couldn't be stored, please check your gas limit.", receipt);
		this.code = ERR_TX_CONTRACT_NOT_STORED;
	}
}

export class TransactionRevertedWithoutReasonError extends TransactionError {
	public constructor(receipt: TransactionReceipt) {
		super(
			`Transaction has been reverted by the EVM:\n ${JSON.stringify(receipt, undefined, 2)}`,
			receipt,
		);
		this.code = ERR_TX_REVERT_WITHOUT_REASON;
	}
}

export class TransactionOutOfGasError extends TransactionError {
	public constructor(receipt: TransactionReceipt) {
		super(
			`Transaction ran out of gas. Please provide more gas:\n ${JSON.stringify(
				receipt,
				undefined,
				2,
			)}`,
			receipt,
		);
		this.code = ERR_TX_OUT_OF_GAS;
	}
}

export class UndefinedRawTransactionError extends TransactionError {
	public constructor() {
		super(`Raw transaction undefined`);
		this.code = ERR_RAW_TX_UNDEFINED;
	}
}
export class TransactionNotFound extends TransactionError {
	public constructor() {
		super('Transaction not found');
		this.code = ERR_TX_NOT_FOUND;
	}
}
