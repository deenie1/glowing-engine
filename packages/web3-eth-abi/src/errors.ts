/*
This file is part of web3.js.

web3.js is free software: you can redistribute it and/or modify
it under the terms of the GNU Lesser General Public License as published by
the Free Software Foundation, either version 3 of the License, or
(at your option) any later version.

web3.js is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU Lesser General Public License for more details.

You should have received a copy of the GNU Lesser General Public License
along with web3.js.  If not, see <http://www.gnu.org/licenses/>.
*/
export class AbiCoderError<T extends Record<string, unknown>&{name?: string} = Record<string, never>> extends Error {
    public readonly props: T
  
    public constructor (
      message: string,
      public readonly code: string,
      props?: T
    ) {
      super(message)
  
      this.name = props?.name ?? 'AbiCoderError'
      this.props = props ?? {} as T
    }
  }

export enum AbiCoderErrors {
    INVALID_ARGUMENT = "INVALID_ARGUMENT",
    INVALID_ABI = "INVALID_ABI"
}