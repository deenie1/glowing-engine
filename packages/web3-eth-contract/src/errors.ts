/*
This file is part of web3.js.

web3.js is free software: you can redistribute it and/or modify
it under the terms of the GNU Lesser General Public License as published by
the Free Software Foundation, either version 3 of the License, or
(at your option) any later version.

web3.js is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU Lesser General Public License for more details.

You should have received a copy of the GNU Lesser General Public License
along with web3.js.  If not, see <http://www.gnu.org/licenses/>.
*/

import { TransactionReceipt } from 'web3-common';
import { ERR_CONTRACT, Web3Error } from 'web3-errors';

export class Web3ContractError extends Web3Error {
	public code = ERR_CONTRACT;
	public receipt?: TransactionReceipt;

	public constructor(message: string, receipt?: TransactionReceipt) {
		super(message);

		this.receipt = receipt;
	}
}
