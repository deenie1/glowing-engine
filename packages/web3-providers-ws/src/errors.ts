import { Web3Error, ERR_PROVIDER } from 'web3-common';

export class Web3WSProviderError extends Web3Error {
	public code = ERR_PROVIDER;
}
