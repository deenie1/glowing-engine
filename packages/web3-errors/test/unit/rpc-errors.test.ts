/*
This file is part of web3.js.

web3.js is free software: you can redistribute it and/or modify
it under the terms of the GNU Lesser General Public License as published by
the Free Software Foundation, either version 3 of the License, or
(at your option) any later version.

web3.js is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU Lesser General Public License for more details.

You should have received a copy of the GNU Lesser General Public License
along with web3.js.  If not, see <http://www.gnu.org/licenses/>.
*/

import { JsonRpcIdentifier } from 'web3-types';

import * as rpcErrors from '../../src/errors/rpc_errors';
import { RpcErrorMessages } from '../../src/errors/rpc_error_messages';

const rpcReturnedError = {
	id: 1,
	jsonrpc: '2.0' as JsonRpcIdentifier,
	error: { code: 123, message: 'error message', data: { a: '10', b: '20' } },
};

describe('rpc errors', () => {
	describe('rpcErrorsMap', () => {
		it('child RpcError classes should be also mapped inside `rpcErrorsMap`', () => {
			expect(
				Object.values(rpcErrors).filter(
					err =>
						(err as any).prototype?.constructor?.prototype instanceof
						rpcErrors.RpcError,
				),
			).toHaveLength(rpcErrors.rpcErrorsMap.size);
		});
	});

	describe('rpcErrors.RpcError', () => {
		it('test constructor', () => {
			expect(new rpcErrors.RpcError(rpcReturnedError).toJSON()).toMatchSnapshot();
		});
		it('test constructor with custom message', () => {
			expect(
				new rpcErrors.RpcError(rpcReturnedError, `some message`).toJSON(),
			).toMatchSnapshot();
		});
	});

	describe('EIP1193ProviderRpcError', () => {
		it('test constructor with no code', () => {
			expect(
				new rpcErrors.EIP1193ProviderRpcError(undefined as unknown as number).toJSON(),
			).toMatchSnapshot();
		});
		it('test constructor with a known code (1000)', () => {
			const code = 1000;
			const error = new rpcErrors.EIP1193ProviderRpcError(code);
			expect(error.toJSON()).toMatchSnapshot();
			expect(error.message).toEqual(RpcErrorMessages[code].message);
		});
		it('test constructor with un registered code', () => {
			expect(new rpcErrors.EIP1193ProviderRpcError(99999).toJSON()).toMatchSnapshot();
		});
	});

	describe('ParseError', () => {
		it('test constructor', () => {
			expect(new rpcErrors.ParseError(rpcReturnedError).toJSON()).toMatchSnapshot();
		});
	});

	describe('InvalidRequestError', () => {
		it('test constructor', () => {
			expect(new rpcErrors.InvalidRequestError(rpcReturnedError).toJSON()).toMatchSnapshot();
		});
	});

	describe('MethodNotFoundError', () => {
		it('test constructor', () => {
			expect(new rpcErrors.MethodNotFoundError(rpcReturnedError).toJSON()).toMatchSnapshot();
		});
	});

	describe('InvalidParamsError', () => {
		it('test constructor', () => {
			expect(new rpcErrors.InvalidParamsError(rpcReturnedError).toJSON()).toMatchSnapshot();
		});
	});

	describe('InternalError', () => {
		it('test constructor', () => {
			expect(new rpcErrors.InternalError(rpcReturnedError).toJSON()).toMatchSnapshot();
		});
	});

	describe('InvalidInputError', () => {
		it('test constructor', () => {
			expect(new rpcErrors.InvalidInputError(rpcReturnedError).toJSON()).toMatchSnapshot();
		});
	});

	describe('MethodNotSupported', () => {
		it('test constructor', () => {
			expect(new rpcErrors.MethodNotSupported(rpcReturnedError).toJSON()).toMatchSnapshot();
		});
	});

	describe('ResourceUnavailableError', () => {
		it('test constructor', () => {
			expect(
				new rpcErrors.ResourceUnavailableError(rpcReturnedError).toJSON(),
			).toMatchSnapshot();
		});
	});

	describe('ResourcesNotFoundError', () => {
		it('test constructor', () => {
			expect(
				new rpcErrors.ResourcesNotFoundError(rpcReturnedError).toJSON(),
			).toMatchSnapshot();
		});
	});

	describe('VersionNotSupportedError', () => {
		it('test constructor', () => {
			expect(
				new rpcErrors.VersionNotSupportedError(rpcReturnedError).toJSON(),
			).toMatchSnapshot();
		});
	});

	describe('TransactionRejectedError', () => {
		it('test constructor', () => {
			expect(
				new rpcErrors.TransactionRejectedError(rpcReturnedError).toJSON(),
			).toMatchSnapshot();
		});
	});

	describe('LimitExceededError', () => {
		it('test constructor', () => {
			expect(new rpcErrors.LimitExceededError(rpcReturnedError).toJSON()).toMatchSnapshot();
		});
	});
});
