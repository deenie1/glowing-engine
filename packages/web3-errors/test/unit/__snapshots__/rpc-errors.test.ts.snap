// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rpc errors EIP1193ProviderRpcError test constructor with a known code (1000) 1`] = `
Object {
  "code": 1000,
  "innerError": undefined,
  "message": "The connection successfully completed the purpose for which it was created.",
  "name": "EIP1193ProviderRpcError",
}
`;

exports[`rpc errors EIP1193ProviderRpcError test constructor with no code 1`] = `
Object {
  "code": undefined,
  "innerError": undefined,
  "message": "",
  "name": "EIP1193ProviderRpcError",
}
`;

exports[`rpc errors EIP1193ProviderRpcError test constructor with un registered code 1`] = `
Object {
  "code": 99999,
  "innerError": undefined,
  "message": "An Rpc error has occured with a code of 99999",
  "name": "EIP1193ProviderRpcError",
}
`;

exports[`rpc errors InternalError test constructor 1`] = `
Object {
  "code": -32603,
  "error": Object {
    "code": 123,
    "data": Object {
      "a": "10",
      "b": "20",
    },
    "message": "error message",
  },
  "id": 1,
  "innerError": undefined,
  "jsonRpc": "2.0",
  "message": "Internal error",
  "name": "InternalError",
}
`;

exports[`rpc errors InvalidInputError test constructor 1`] = `
Object {
  "code": -32000,
  "error": Object {
    "code": 123,
    "data": Object {
      "a": "10",
      "b": "20",
    },
    "message": "error message",
  },
  "id": 1,
  "innerError": undefined,
  "jsonRpc": "2.0",
  "message": "Invalid input",
  "name": "InvalidInputError",
}
`;

exports[`rpc errors InvalidParamsError test constructor 1`] = `
Object {
  "code": -32602,
  "error": Object {
    "code": 123,
    "data": Object {
      "a": "10",
      "b": "20",
    },
    "message": "error message",
  },
  "id": 1,
  "innerError": undefined,
  "jsonRpc": "2.0",
  "message": "Invalid params",
  "name": "InvalidParamsError",
}
`;

exports[`rpc errors InvalidRequestError test constructor 1`] = `
Object {
  "code": -32600,
  "error": Object {
    "code": 123,
    "data": Object {
      "a": "10",
      "b": "20",
    },
    "message": "error message",
  },
  "id": 1,
  "innerError": undefined,
  "jsonRpc": "2.0",
  "message": "Invalid request",
  "name": "InvalidRequestError",
}
`;

exports[`rpc errors LimitExceededError test constructor 1`] = `
Object {
  "code": -32005,
  "error": Object {
    "code": 123,
    "data": Object {
      "a": "10",
      "b": "20",
    },
    "message": "error message",
  },
  "id": 1,
  "innerError": undefined,
  "jsonRpc": "2.0",
  "message": "Limit exceeded",
  "name": "LimitExceededError",
}
`;

exports[`rpc errors MethodNotFoundError test constructor 1`] = `
Object {
  "code": -32601,
  "error": Object {
    "code": 123,
    "data": Object {
      "a": "10",
      "b": "20",
    },
    "message": "error message",
  },
  "id": 1,
  "innerError": undefined,
  "jsonRpc": "2.0",
  "message": "Method not found",
  "name": "MethodNotFoundError",
}
`;

exports[`rpc errors MethodNotSupported test constructor 1`] = `
Object {
  "code": -32004,
  "error": Object {
    "code": 123,
    "data": Object {
      "a": "10",
      "b": "20",
    },
    "message": "error message",
  },
  "id": 1,
  "innerError": undefined,
  "jsonRpc": "2.0",
  "message": "Method not supported",
  "name": "MethodNotSupported",
}
`;

exports[`rpc errors ParseError test constructor 1`] = `
Object {
  "code": -32700,
  "error": Object {
    "code": 123,
    "data": Object {
      "a": "10",
      "b": "20",
    },
    "message": "error message",
  },
  "id": 1,
  "innerError": undefined,
  "jsonRpc": "2.0",
  "message": "Parse error",
  "name": "ParseError",
}
`;

exports[`rpc errors ResourceUnavailableError test constructor 1`] = `
Object {
  "code": -32002,
  "error": Object {
    "code": 123,
    "data": Object {
      "a": "10",
      "b": "20",
    },
    "message": "error message",
  },
  "id": 1,
  "innerError": undefined,
  "jsonRpc": "2.0",
  "message": "Resource unavailable",
  "name": "ResourceUnavailableError",
}
`;

exports[`rpc errors ResourcesNotFoundError test constructor 1`] = `
Object {
  "code": -32001,
  "error": Object {
    "code": 123,
    "data": Object {
      "a": "10",
      "b": "20",
    },
    "message": "error message",
  },
  "id": 1,
  "innerError": undefined,
  "jsonRpc": "2.0",
  "message": "Resource not found",
  "name": "ResourcesNotFoundError",
}
`;

exports[`rpc errors TransactionRejectedError test constructor 1`] = `
Object {
  "code": -32003,
  "error": Object {
    "code": 123,
    "data": Object {
      "a": "10",
      "b": "20",
    },
    "message": "error message",
  },
  "id": 1,
  "innerError": undefined,
  "jsonRpc": "2.0",
  "message": "Transaction rejected",
  "name": "TransactionRejectedError",
}
`;

exports[`rpc errors VersionNotSupportedError test constructor 1`] = `
Object {
  "code": -32006,
  "error": Object {
    "code": 123,
    "data": Object {
      "a": "10",
      "b": "20",
    },
    "message": "error message",
  },
  "id": 1,
  "innerError": undefined,
  "jsonRpc": "2.0",
  "message": "JSON-RPC version not supported",
  "name": "VersionNotSupportedError",
}
`;

exports[`rpc errors rpcErrors.RpcError test constructor 1`] = `
Object {
  "code": 123,
  "error": Object {
    "code": 123,
    "data": Object {
      "a": "10",
      "b": "20",
    },
    "message": "error message",
  },
  "id": 1,
  "innerError": undefined,
  "jsonRpc": "2.0",
  "message": "An Rpc error has occured with a code of 123",
  "name": "RpcError",
}
`;

exports[`rpc errors rpcErrors.RpcError test constructor with custom message 1`] = `
Object {
  "code": 123,
  "error": Object {
    "code": 123,
    "data": Object {
      "a": "10",
      "b": "20",
    },
    "message": "error message",
  },
  "id": 1,
  "innerError": undefined,
  "jsonRpc": "2.0",
  "message": "some message",
  "name": "RpcError",
}
`;
